synapse/storage/databases/main/push_rule.py

```diff
@@ -174,7 +145,7 @@ def get_max_push_rules_stream_id(self) -> int:
         raise NotImplementedError()
 
     @cached(max_entries=5000)
-    async def get_push_rules_for_user(self, user_id: str) -> List[JsonDict]:
+    async def get_push_rules_for_user(self, user_id: str) -> FilteredPushRules:
```

Comment:
Do we still need to do the sorting by priority class / priority here? (Isn't this handled by `PushRules` now?)

Reply:
The `PushRules` class will correctly sort by priority class, but doesn't have a concept of "priority" within a class, so expects to get the push rules in the correct order